From 4bd2bb45d605feeffb5c316c562956c3c4c63a62 Mon Sep 17 00:00:00 2001
From: Guy Nisan <gnisan@habana.ai>
Date: Tue, 29 Oct 2024 15:43:25 +0200
Subject: [PATCH] [FSW-7732] package: agent-fwl: skip bin install

Since FS1 agents do not produce binary out of elf as part of zephyr
build process, skip bin install if file does not exists.

Change-Id: Id4a243b4e51a4978f89b4b8e5d0889dffd04c90e
Signed-off-by: Guy Nisan <gnisan@habana.ai>
---

diff --git a/package/agent-fwl/agent-fwl.mk b/package/agent-fwl/agent-fwl.mk
index fe7d6b0..3aa1f5f 100644
--- a/package/agent-fwl/agent-fwl.mk
+++ b/package/agent-fwl/agent-fwl.mk
@@ -134,7 +134,9 @@
 endef
 
 define AGENT_FWL_INSTALL_TARGET_CMDS
-	@$(INSTALL) -D -m 0644 $(@D)/build/zephyr/zephyr.bin $(BINS_DIR)/zephyr_agent_$(AGENT_FWL_APP_NAME).bin
+	@if [ -f $(@D)/build/zephyr/zephyr.bin ]; then \
+		$(INSTALL) -D -m 0644 $(@D)/build/zephyr/zephyr.bin $(BINS_DIR)/zephyr_agent_$(AGENT_FWL_APP_NAME).bin; \
+	fi
 	@$(INSTALL) -D -m 0644 $(@D)/build/zephyr/zephyr.elf $(ELFS_DIR)/zephyr_agent_$(AGENT_FWL_APP_NAME).elf
 	@$(INSTALL) -D -m 0644 $(@D)/build/zephyr/zephyr.stat $(STATS_DIR)/zephyr_agent_$(AGENT_FWL_APP_NAME).stat
 
